arches/app/media/js/viewmodels/workflow-step.js

```diff
@@ -227,6 +236,10 @@ define([
             }
         }
 
+        this.setSourceStepLock = function(locked) {
+            config.workflow.setLock(self.externalStepData.sourcesteptolock.stepName, locked);
```

Comment:
ah, one small thing. Otherwise everything looks good!

`externalStepData` is used for passing data in one direction from one step to another.
Can the config key be something separate? Something like `lockableExternalSteps` ?